skimage/filters/thresholding.py

```diff
@@ -384,3 +385,137 @@ def threshold_li(image):
             new_thresh = temp + tolerance
 
     return threshold + immin
+
+
+def threshold_multiotsu(image, nclass=3, nbins=256):
+    """Generates multiple thresholds for an input image. Based on the
+    Multi-Otsu implementation by Liao and Chung.
+
+    Parameters
+    ----------
+    image : array
+        Grayscale input image.
+    nclass : int, optional
+        Number of classes to be thresholded, i.e. the number of resulting
+        regions. Accepts an integer from 2 to 5.
+    nbins : int, optional
+        Number of bins used to calculate histogram. This value is ignored for
+        integer arrays.
+
+    Returns
+    -------
+    idx_thresh : array
+        Array containing the threshold values for the desired classes.
+
+    References
+    ----------
+    .. [1] Liao, P-S. and Chung, P-C., "A fast algorithm for multilevel
+    thresholding", Journal of Information Science and Engineering 17
+    (5): 713-727, 2001.
+
+    Examples
+    --------
+    >>> from skimage.data import camera
+    >>> image = camera()
+    >>> thresh = threshold_multiotsu(image)
+    >>> region1 = image <= thresh[0]
+    >>> region2 = (image > thresh[0]) & (image <= thresh[1])
+    >>> region3 = image > thresh[1]
+    """
+
+    # check if the image is RGB.
+    if image.shape[-1] in (3, 4):
+        raise TypeError("Your image seems to be RGB (shape: {0}. Please use a"
+                        "grayscale image.".format(image.shape))
+
+    # check if the image has more than one color.
+    if image.min() == image.max():
+        raise TypeError("The input image seems to have just one color: {0}."
+                        "Please use a grayscale image.".format(image.min()))
+
+    # image needs to be treated as float.
+    image = img_as_float(image)
+
+    # calculating the histogram and the probability of each gray level.
+    hist, _ = histogram(image.ravel(), nbins)
+    prob = hist/np.size(image)
+
+    gray_lvl = 256
```

Comment:
@sciunto I could not solve this problem. I tried to reproduce the behavior of Tosa's plugin: http://imagej.net/plugins/download/Multi_OtsuThreshold.java
However, I didn't figure out yet heow to divide the classes according to the kind of the input image... for now, this code is useless.